src/containers/trends/CBGTrendsContainer.js

```diff
@@ -66,14 +65,56 @@ class CBGTrendsContainer extends React.Component {
       targetLowerBound: PropTypes.number.isRequired,
       veryLowThreshold: PropTypes.number.isRequired,
     }),
-    bgUnits: PropTypes.oneOf(['mg/dL', 'mmol/L']),
-    data: PropTypes.array.isRequired,
-    focusedSlice: PropTypes.object,
-    focusedSliceKeys: PropTypes.array,
+    bgUnits: PropTypes.oneOf(['mg/dL', 'mmol/L']).isRequired,
+    data: PropTypes.arrayOf(PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      msPer24: PropTypes.number.isRequired,
+      value: PropTypes.number.isRequired,
+    })).isRequired,
+    focusedSlice: PropTypes.shape({
+      slice: PropTypes.shape({
+        firstQuartile: PropTypes.number.isRequired,
+        max: PropTypes.number.isRequired,
+        median: PropTypes.number.isRequired,
+        min: PropTypes.number.isRequired,
+        ninetiethQuantile: PropTypes.number.isRequired,
+        tenthQuantile: PropTypes.number.isRequired,
+        thirdQuartile: PropTypes.number.isRequired,
+      }).isRequired,
+      position: PropTypes.shape({
+        left: PropTypes.number.isRequired,
+        tooltipLeft: PropTypes.bool.isRequired,
+        topOptions: PropTypes.shape({
+          firstQuartile: PropTypes.number.isRequired,
+          max: PropTypes.number.isRequired,
+          median: PropTypes.number.isRequired,
+          min: PropTypes.number.isRequired,
+          ninetiethQuantile: PropTypes.number.isRequired,
+          tenthQuantile: PropTypes.number.isRequired,
+          thirdQuartile: PropTypes.number.isRequired,
+        }).isRequired,
+      }).isRequired,
+    }),
+    focusedSliceKeys: PropTypes.arrayOf(PropTypes.oneOf([
+      'firstQuartile',
```

Comment:
I actually want @krystophv's opinion on this. Because if we make this a constant (e.g., `const FIRST_QUARTILE = 'firstQuartile'`, we should also define the propTypes like so:
```
focusedSlice: PropTypes.shape({
   [FIRST_QUARTILE]: PropTypes.number.isRequired,
```
That will be a LOT of changes and it seems quite obscuring/opaque to me. Plus with tab completion *and* all the propTypes defined in full in every component, it's kinda hard to misspell things because the tab completion should grab it from the propTypes.